### PR TITLE
F/U download host fallback

### PR DESCRIPTION
**Pull request by ChrisSchinnerl**
This addresses a few things I noticed while looking at the download code for the host fallback code.

### CODE REVIEW

File: worker/download.go [hunk 1]
```diff
@@ -755,12 +755,9 @@ loop:
...
 			// handle errors
 			if resp.err != nil {
-				// launch overdrive requests
-				for {
-					if req := s.nextRequest(ctx, resps, true); req != nil {
-						s.launch(req)
-					}
-					break
+				// launch replacement request
+				if req := s.nextRequest(ctx, resps, resp.req.overdrive); req != nil {
```
Comment (ChrisSchinnerl) [inline worker/download.go:759]: What seemed odd to me here is that we always consider a launched job an overdrive job even if it is just a replacement for a non-overdrive request.

File: worker/download.go [hunk 2]
```diff
@@ -611,8 +611,8 @@ func (s *slabDownload) overdrive(ctx context.Context, resps *sectorResponses) (r
 			case <-ctx.Done():
 				return
 			case <-timer.C:
-				if canOverdrive(timeout()) {
-					for {
+				for {
+					if canOverdrive(timeout()) {
```
Comment (ChrisSchinnerl) [inline worker/download.go:615]: Felt right to move this inside since every launch of a job affects the outcome of `canOverdrive`.

Comment (peterjan) [reply worker/download.go:615]: Yup, makes sense I think

File: worker/download.go (hunk 1 above)
Comment (peterjan) [reply worker/download.go:759]: It depends on the definition of an overdrive request. Both are fine I think, this is probably better though. It's likely introduced by a small refactor at some point and overlooked until now.

**Review by peterjan — APPROVED** [review]

**Comment by peterjan** [comment]
Taking this along in https://github.com/SiaFoundation/renterd/pull/1587

**Pull request closed without merging**